packages/sdk/src/react/_internal/transaction-machine/execute-transaction.ts

```diff
@@ -328,13 +331,21 @@ export class TransactionMachine {
 					break;
 
 				case TransactionType.OFFER:
+					// for fetching steps for making offer, we need to check if spending erc20 token is needed, so just for first fetch we set pricePerToken to 1
+					const offer = !this.transactionState.approval.checked
+						? {
+								...props.offer,
+								pricePerToken: '1',
+							}
+						: props.offer;
+
```

Comment:
I dont think this is correct to have here. We used it in the modal to be able to check if the user has at least approved 1 token to be transferred. That way we avoided to add and remove the approve button for the most common case, causing the modal to flicker and resize.
We still need to check if the user has approved the amount he is offering